src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php

```diff
@@ -100,6 +100,13 @@ public static function createResourceForClass(\ReflectionClass $reflectionClass)
      */
     protected function processValue($value, $isRoot = false)
     {
+        if ($value instanceof Reference && !$this->container->has($id = (string) $value)) {
+            if (preg_match('/^[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*+(?:\\\\[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*+)++$/', $id) && $reflectionClass = $this->container->getReflectionClass($id, true)) {
+                $this->createAutowiredDefinition($reflectionClass);
+            }
+
+            return parent::processValue($value, $isRoot);
```

Comment:
no practical side effect indeed, because we know that the parent does nothing, but still useful to *not* have to know about that - ie for locality - a core robustness principle to me.